### PR TITLE
Round and style buttons on close all tabs dialog

### LINKED ISSUE #3379: "Close all tabs?" dialog should highlight the "default" option
refer to screenshot:

image

Pressing enter here _closes the window_, because that's the focused/default button on this dialog. However, there's no way of knowing which is the selected button right now. We should either use the accent color with the button, or manually display the focus rect (is that even possible)?

Also for discussion: Should "close" be the default option here?

### LINKED ISSUE #4307: Button corners in the "Do you want to close all tabs" pop-up dialogue are not rounded
# Environment

```none
Windows build number: 10.0.19041.0
Windows Terminal version (if applicable): 0.8.10091.0

Any other software? No
```

# Steps to reproduce
Open at least two tabs in Windows Terminal and press the close button

# Expected behavior
The grey buttons in the "Do you want to close all tabs" pop-up dialogue should have rounded corners from the latest WinUI library, like in the "About" pop-up, for better consistency

# Actual behavior
These grey buttons don't have rounded corners (like in older WinUI versions), which is weird because Windows Terminal uses the latest WinUI version & elements

### PR DESCRIPTION
**Pull request by cinnamon-msft**
## Summary of the Pull Request
Added styling to the buttons on the Close All Tabs dialog box.

 
## PR Checklist
* [x] Closes #4307
* [x] Closes #3379  
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
image

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -284,15 +284,15 @@ namespace winrt::TerminalApp::implementation
...
         auto title = RS_(L"CloseWindowWarningTitle");
-        auto primaryButtonText = RS_(L"CloseAll");
-        auto secondaryButtonText = RS_(L"Cancel");
+        auto closeButtonText = RS_(L"CloseAll");
+        auto primaryButtonText = RS_(L"Cancel");
 
         WUX::Controls::ContentDialog dialog;
         dialog.Title(winrt::box_value(title));
 
+        dialog.CloseButtonText(closeButtonText);
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: wow, we were using the wrong button!

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: I am actually somewhat concerned about this. :)
What if you press _escape_ while the dialog is open?

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: > Use the CloseButtonText and property to set the display text for the safe, non-destructive button. The dialog's close button will also be invoked when the user performs a Cancel action, like pressing the ESC key or pressing the system back button. Use the PrimaryButtonText and SecondaryButtonText properties to display responses to the main question or action posed by the dialog.

big bad news

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: the close button is for closing the dialog, not for closing the application. :smile:

Comment (cinnamon-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Should I leave it as the primary button but style it with the active coloring? (same as how we styled the close button)

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 2]
```diff
@@ -284,15 +286,16 @@ namespace winrt::TerminalApp::implementation
     void TerminalPage::_ShowCloseWarningDialog()
     {
         auto title = RS_(L"CloseWindowWarningTitle");
-        auto primaryButtonText = RS_(L"CloseAll");
-        auto secondaryButtonText = RS_(L"Cancel");
+        auto closeButtonText = RS_(L"CloseAll");
+        auto primaryButtonText = RS_(L"Cancel");
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: this is still incorrect. The "close" button _is meant to exit the dialog_. The "primary" button is meant to _activate the dialog's action._.

Comment (cinnamon-msft) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: So the default button should be Cancel instead of CloseAll?

Comment (DHowett-MSFT) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: primary=close all, close=cancel, default=primary.

**Pull request merged**